indico/modules/events/persons/__init__.py

```diff
@@ -49,8 +51,15 @@ def _get_placeholders(sender, person, event, contribution=None, abstract=None, r
         yield person_placeholders.ContributionsPlaceholder
 
 
+class CustomPersonsMode(RichIntEnum):
+    __titles__ = [None, _('Always'), _('After search'), _('Never')]
+    always = 1
+    after_search = 2
+    never = 3
+
+
 persons_settings = EventSettingsProxy('persons', {
-    'disallow_custom_persons': False,  # Disallow manually entering persons on person lists
+    'custom_persons_mode': CustomPersonsMode.always,  # Prevent submitters from adding people manually before searching
```

Comment:
I'd move the documentation for the behavior into the enum - there you can nicely explain what each option does, instead of having to squeeze something very short into half a line